Cargo.toml

```diff
@@ -8,11 +8,103 @@ codegen-units = 1
 panic = "abort"
 lto = "fat"
 
+# Copied from https://github.com/paradigmxyz/reth/blob/bd8c4eceb20c39c6e501d06cf906469329340bb9/Cargo.toml#L159
```

Comment:
This copy-pasting from Reth is pretty noisy. Let's only keep things that are relevant for now: 
- `rust.rust_2018_idioms = "warn"` for `[workspace.lints]`, and:
```
[workspace.lints.clippy]
cloned_instead_of_copied = "warn"
derive_partial_eq_without_eq = "warn"
doc_markdown = "warn"
explicit_iter_loop = "warn"
if_not_else = "warn"
if_then_some_else_none = "warn"
match_same_arms = "warn"
missing_const_for_fn = "warn"
redundant_clone = "warn"
redundant_else = "warn"
type_repetition_in_bounds = "warn"
unnested_or_patterns = "warn"
use_self = "warn"
```